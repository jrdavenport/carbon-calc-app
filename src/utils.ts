export interface transportData {
  schoolName: string,
  className: string,
  zone: {
    animal: string;
    colour: string;
  };
  transport: string;
  date: string;
  journey: string;
}

export interface buttonValue {
  value: string;
  img?: string;
  colour?: string;
}

export interface school {
  name: string,
  classes: string[],
  zones: {
    animals: {
      value: string,
      img: string,
    }[],
    colours:
    {
      value: string,
      colour: string,
    }[],
  },
  transports: {
    value: string,
    img: string,
  }[],
}

export interface selectedClass {
  schoolName: string,
  className: string,
  zones: {
    animals: {
      value: string,
      img: string,
    }[],
    colours:
    {
      value: string,
      colour: string,
    }[],
  },
  transports: {
    value: string,
    img: string,
  }[],
}

export const transportStorageKey = "transportData";
export const schoolStorageKey = "schoolData";

const toLowerSnakeCase = (string: string) =>
  string.replace(" ", "_").toLowerCase();

export const adaptRecords = (records: Array<transportData>) => {
  console.log("recs", records);
  return records.map(({ journey, date, zone, transport }, index) => ({
    pupil_id: index,
    class_id: 1,
    school_id: 1,
    date,
    journey: journey.toLowerCase(),
    shares: 0,
    method: toLowerSnakeCase(transport),
    zone_animal: toLowerSnakeCase(zone.animal),
    zone_colour: toLowerSnakeCase(zone.colour),
  }));
};
