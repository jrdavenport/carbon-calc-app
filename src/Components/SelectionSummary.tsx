import { Grid, Typography } from "@material-ui/core";
import React from "react";
import { buttonValue } from "../utils";
import SelectionButton from "./SelectionButton";

interface Props {
  zoneAnimal: buttonValue;
  zoneColour: buttonValue;
  transport: buttonValue;
}

function SelectionSummary({ zoneAnimal, zoneColour, transport }: Props) {
  return (
    <Grid container spacing={3} justify="center">
      <Grid item xs={12}>
        <Typography variant="h3" gutterBottom>
          Please confirm your selection:
        </Typography>
      </Grid>
      <Grid item xs={12} sm={3}>
        <SelectionButton
          color={zoneColour.colour}
          transport={transport.img}
          animal={zoneAnimal.img}
          onClick={() => { }}
          text={{ zone: zoneAnimal.value, transport: transport.value, zoneColour: zoneColour.value }}
        />
      </Grid>
    </Grid>
  );
}

export default SelectionSummary;
