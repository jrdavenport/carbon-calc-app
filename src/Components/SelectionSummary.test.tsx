import { render } from "@testing-library/react";
import React from "react";
import { buttonValue } from "../utils";
import SelectionSummary from "./SelectionSummary";

const animal: buttonValue = {
  "value": "Dog",
  "img": "🐶"
}

const colour: buttonValue = {
  "value": "Red",
  "colour": "LightCoral"
}

const transport: buttonValue = {
  "value": "Train",
  "img": "🚆"
}


const setup = () => {
  const { container } = render(<SelectionSummary
    zoneAnimal={animal}
    zoneColour={colour}
    transport={transport} />);
  return {
    container,
  };
};

it('matches snapshot', () => {
  const { container } = setup();

  expect(container).toMatchSnapshot();
})
