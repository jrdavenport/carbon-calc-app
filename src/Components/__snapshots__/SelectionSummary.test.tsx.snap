// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom"
      >
        Please confirm your selection:
      </h3>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-outlined MuiPaper-rounded"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root makeStyles-boarder-4"
          tabindex="0"
          type="button"
        >
          <div
            class="MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
              >
                Red Dog travelling by Train
              </span>
            </div>
          </div>
          <div
            class="MuiCardContent-root"
            style="background-color: LightCoral;"
          >
            <div
              class="MuiTypography-root makeStyles-smallEmoji-3 MuiTypography-body1"
            >
              üê∂
            </div>
            <div
              class="MuiTypography-root makeStyles-smallEmoji-3 MuiTypography-body1"
            >
              üöÜ
            </div>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
